### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by dbasilio**
Initial implementation.

### CODE REVIEW

File: .semver-up.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "packagePrefix": "eslint"
+}
```
Comment (dbasilio) [inline .semver-up.json:3]: Just for testing purposes

File: src/index.ts [hunk 2]
```diff
@@ -23,13 +49,79 @@ class SemverUpCommand extends Command<CommandContext> {
...
             this.context.stdout.write('Done\n')
             return 0
         } catch (err) {
             this.context.stderr.write(`${String(err)}\n`)
             return 1
         }
     }
+
+    async parseConfigFile(): Promise<Config> {
+        const configFile = this.configFile ? this.configFile : '.semver-up.json'
```
Comment (dbasilio) [inline src/index.ts:67]: Allows for users to change the name of the config file from the default

File: src/index.ts [hunk 3]
```diff
@@ -23,13 +49,79 @@ class SemverUpCommand extends Command<CommandContext> {
...
+
+    async getUpdateCandidates(
+        config: Config,
+        manifest: Manifest,
+    ): Promise<Descriptor[]> {
+        const candidates: Descriptor[] = []
+        const possiblePackages = [
+            ...manifest.dependencies.values(),
+            ...manifest.devDependencies.values(),
+        ]
```
Comment (dbasilio) [inline src/index.ts:108]: Not sure if peer dependencies should be considered here or not

**Comment by noahnu** [comment]
@all-contributors add dbasilio for idea, code

**Comment by noahnu** [comment]
@all-contributors add noahnu for code,infra

**Comment by noahnu** [comment]
@all-contributors add noahnu for code, infra

**Pull request merged**